src/system.hpp

```diff
@@ -102,3 +181,11 @@ const size_t READ_BUFFER_SIZE = 65536;
 #else
 #  define IS_TRIVIALLY_COPYABLE(T) std::is_trivially_copyable<T>::value
 #endif
+
+#ifdef max
```

Comment:
Maybe this is easier? (before the includes, specifically Windows.h / Windef.h)
#define NOMINMAX 1